### PR TITLE
Core: Gardening updates from Eden

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [not yet] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

I was poking around open PRs and I saw we have some stuff about gardening, so I went and did some comparisons with Eden. I clarified some comments and took what changes I noticed by eye

### CODE REVIEW

File: src/map/utils/gardenutils.cpp [hunk 1]
```diff
@@ -275,7 +278,14 @@ namespace gardenutils
             strength += dominantAura / 10;
         }
 
-        strength += (int16)((100 - strength) * (PItem->getStrength() / 32.0f));
+        strength += (int16)((100 - strength) * (PItem->getStrength() / 31.0f));
```
Comment (lapislosh) [inline src/map/utils/gardenutils.cpp:281]: I have a previous PR that fixes this a different way: https://github.com/LandSandBoat/server/pull/4522
I chose to increase the RNG to 0-32 and keep the divisor as 32, I guess they chose to drop the divisor to 31 but leave the range from 0-31.  Either is probably fine tbh but we can't have both changes together or it'll be wrong.

**Comment by lapislosh** [comment]
Also for completeness sake, I had removed double ore harvests from my previous PR with the comment:

> If we did want to keep the maximum as 2 but make it super rare, we would need to adjust the gardening_results.sql to have multiple entries for each ore, and only the last entry would have a yield of 2.

This PR makes this change, and so double-ore harvests will now be possible again (and it's done the correct way here with 2 separate entries to make it rare, instead of having one entry with a range of 1-2).

File: src/map/utils/gardenutils.cpp (hunk 1 above)
Comment (zach2good) [reply src/map/utils/gardenutils.cpp:281]: Thanks for weighing in, I was just pulling things over willy-nilly 👍

**Comment by zach2good** [comment]
I've abandoned this PR. Downstream has apparently been testing and trying to fix some of the small bugs in gardening, so hopefully they'll eventually find their way back upstream one day

**Pull request closed without merging**